src/swap/components/SwapSettingsSlippageInput.tsx

```diff
@@ -0,0 +1,103 @@
+import { useState } from 'react';
+import { background, border, cn, color, pressable } from '../../styles/theme';
+import type { SwapSettingsSlippageInputReact } from '../types';
+import { useSwapContext } from './SwapProvider';
+
+export function SwapSettingsSlippageInput({
+  className,
+  defaultSlippage = 3,
+}: SwapSettingsSlippageInputReact) {
+  const { setLifeCycleStatus } = useSwapContext();
+  const [slippage, setSlippage] = useState(defaultSlippage);
+  const [isAutoMode, setIsAutoMode] = useState(true);
+
+  const updateSlippage = (newSlippage: number) => {
+    setSlippage(newSlippage);
+    setLifeCycleStatus({
+      statusName: 'slippageChange',
+      statusData: { maxSlippage: newSlippage },
+    });
+  };
+
+  const handleSlippageChange = (newSlippage: string) => {
```

Comment:
should all handle be behind a useCallack.

Read this https://coderpad.io/blog/development/a-guide-to-using-reacts-usecallback-hook/ so you better understand the re-render dynamics of React.

Also re-watch this video https://www.youtube.com/watch?v=-KVclgnfGms and trully internalize the meaning behind React component rendering.

Reply:
Memoizing `updateSlippage`, `handleSlippageChange`, and `handleSlippageSettingChange`